### PR TITLE
Fix mixed ES syntax

### PR DESCRIPTION
**Pull request by alexmiranda**
## What's changed

* Moved `index.js` into `src` so it's transpiled as the other files
* Changed main file to `./lib/index.js`
* Converted `index.js` to ES6 
* Added unit tests for `index.js`

Thanks to @paulcodiny for reporting the issue.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  createApp,
+  createComponent,
+  createFactory,
+  createModel,
+  createService,
+  mapToProps,
+  Model,
+  PropTypes,
+  Region,
+  render,
```
Comment (AlexDudar) [inline src/index.js]: not necessary comma

Comment (alexmiranda) [inline src/index.js]: our lint rules require a trailing comma for ES6 code

**Comment by alexmiranda** [comment]
after adding unit tests, the test coverage _decreased_! 😐

File: src/index.js (hunk 1 above)
Comment (AlexDudar) [inline src/index.js]: strange.. can't say that I like it, but fine

Comment (alexmiranda) [inline src/index.js]: dangle commas help a lot when visualising changes and make it more convenient and less error-prone for manual merges.

**Comment by fahad19** [comment]
👍

**Pull request merged**